platform/src/ToolsManager.js

```diff
@@ -52,8 +53,17 @@ class ToolManager {
 
             let xhr = new XMLHttpRequest();
             xhr.open("GET", toolUrl.url, false);
-            xhr.send();
-    
+
+            try{
+                xhr.send();
+            } catch (e) {
+                if (e instanceof DOMException){
+                    errors.push( new EducationPlatformError(`A tool configuration file was not accessible at: ${toolUrl.url}. 
+                                                            Check the tool's url given in the activity file is correct and the tool 
+                                                            service is still available.`) );
+                }
```

Comment:
Would it make sense to `return errors;` here?

Reply:
The suggested change would mean only the first error is displayed on a tool being unreachable when there could be multiple tools that are unreachable. Currently all the unreachable tools are displayed so the teacher will see all the errors at once rather than re-run and then see the next etc. I prefer this behaviour but it could be changed if we want to incrementally show the errors.

Reply:
Understood. I was mainly worried about errors happening in the next few steps because they would still be executed even if the tool configuration couldn't be accessed.